videocore6/assembler.py

```diff
@@ -13,6 +13,12 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.labels = {}
 
+    def _gen_unused_label(self, label_format='{}'):
+        n = len(self.labels)
+        while label_format.format(n) in self.labels:
+            n += 1
+        return label_format.format(n)
```

Comment:
When an evil user specifies a string that does not contain any `{}` to `label_format`, then this loops infinitely since `.format` ignores extra arguments. So I think we should instead have `label_prefix` argument and create labels such that `label_prefix + '_' + str(n)`.

Reply:
At first, I impl like `def _gen_unused_label(self, prefix = None, suffix = None):`, and change to current on for flexibility.

I think a better way to avoid an infinite loop is checking `label_format.format(n)` is fixpoint or not.

Reply:
I think too it is a good idea! Can you fix the code in this way?

Reply:
Done.

And python `_` prefixed symbol means private interface, so there is no problem to ignore a user who uses such a dangerous one.

Reply:
Confirmed to be fixed. Thanks!